src/Agent.Worker/Release/ContainerProvider/Helpers/VssConnectionFactory.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Services.Client;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.OAuth;
+using Microsoft.VisualStudio.Services.WebApi;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release.ContainerProvider
+{
+    /// <summary>
+    /// Our factory for VssConnections. Used to ensure:
+    ///  1. consistent initialization
+    ///  2. Connect() is called, but only once for a given Uri.
```

Comment:
you mean...
Connect() is called, but the connection is created only once for a given Uri. 
?